src/site/resources/styleguides/google-java-style-20180523/javaguide.html

```diff
@@ -278,6 +282,11 @@ <h4 id="s4.1.2-blocks-k-r-style">4.1.2 Nonempty blocks: K &amp; R style</h4>
   <code class="prettyprint lang-java">else</code> or a comma.</li>
 </ul>
 
+<p>Exception: In places where these rules allow a single statement ending with a semicolon
+(<code class="prettyprint lang-java">;</code>), a block of statements can appear, and the opening
+brace of this block is preceded by a line break. Blocks like these are typically introduced to
+limit the scope of local variables, for example inside switch statements.</p>
```

Comment:
@Zopsss , do not hesitate to look at git history of style guide - https://github.com/google/styleguide/commit/49488412b2f59843fce2433bd834a3fd9700604e#diff-10efc97bb07838e0dcb3978a42e46089cca7c09cd37aa2082efc93db2b27ab4fR285
so you can see when it was added :). so we can spot extra example was added.

unfortunately related PR has no details - https://github.com/google/styleguide/pull/671

Reply:
we need to see how we behave on code like:
```
public void method() {
    {
      int x = foo();
      frob(x);
    }
  }
```
or in switch:
```
 case 2: {
      int x = foo();
      frob(x);
  } 
```

Reply:
still a concerning is:
>  In places where these rules allow a single statement ending with a semicolon (;)

it sound like they try to allow single line methods
`public void getX() { return X;}`
but I am not sure.

Reply:
@Zopsss , please become familiar with https://github.com/google/google-java-format , it is maintained by `cushon` user, and he also update Style guide.
We have even issues that our validation is not matching this tool.
Lets see how this tool behave on such single line methods.

Reply:
looks like this is where it is reported - https://github.com/checkstyle/checkstyle/issues/14294#issuecomment-1904385363 attention to user of this post :).

Reply:
we can just put blue tick on this and make comment in coverage table that until https://github.com/checkstyle/checkstyle/issues/14294#issuecomment-1904385363, it not compete coverage, and it will be simply covered as bug fix.

Reply:
I have made example about it on my repo, here's the input file link:

https://github.com/Zopsss/Google-Java-Format-Testing/blob/master/src/main/java/com/examples/nonemptyblocks/Checkstyle.java

I ran the google_checks.xml against it and attached the output at the end of the file with explanation with explanation. 

Here's the diff between the code generated by google-java-format and normal input file against which I ran checkstyle:

https://github.com/Zopsss/Google-Java-Format-Testing/blob/master/src/main/java/com/examples/nonemptyblocks/git.diff

Reply:
> still a concerning is:
> 
> > In places where these rules allow a single statement ending with a semicolon (;)
> 
> it sound like they try to allow single line methods `public void getX() { return X;}` but I am not sure.

The google-java-format tool is converting this code to:

```java
  public int getX() {
    return 0;
  }
```

pls refer to the git.diff file

Reply:
Yes, they meant only blocks in `case`.

Reply:
Will be handled by https://github.com/checkstyle/checkstyle/issues/14294

Reply:
Placed to ToDo queue of project

Reply:
coverage update is at https://github.com/checkstyle/checkstyle/pull/15417 ? why different issue is referenced?

Reply:
fixed the referenced issue.

Reply:
coverage update is provided https://github.com/checkstyle/checkstyle/pull/15417.
This review thread is resolved

Reply:
coverage is update at https://github.com/checkstyle/checkstyle/pull/15417
review thread is resolved.